march_hardware_builder/include/march_hardware_builder/hardware_builder.h

```diff
@@ -66,21 +67,21 @@ class HardwareBuilder
   static void validateRequiredKeysExist(const YAML::Node& config, const std::vector<std::string>& key_list,
                                         const std::string& object_name);
 
-  static march::Joint createJoint(const YAML::Node& joint_config, const std::string& joint_name,
-                                  const urdf::JointConstSharedPtr& urdf_joint, march::PdoInterfacePtr pdo_interface,
-                                  march::SdoInterfacePtr sdo_interface);
+  march::Joint createJoint(const YAML::Node& joint_config, const std::string& joint_name,
+                           const urdf::JointConstSharedPtr& urdf_joint, march::PdoInterfacePtr pdo_interface,
+                           march::SdoInterfacePtr sdo_interface);
   static std::unique_ptr<march::AbsoluteEncoder> createAbsoluteEncoder(const YAML::Node& absolute_encoder_config,
                                                                        const urdf::JointConstSharedPtr& urdf_joint);
   static std::unique_ptr<march::IncrementalEncoder>
   createIncrementalEncoder(const YAML::Node& incremental_encoder_config);
-  static std::unique_ptr<march::IMotionCube> createIMotionCube(const YAML::Node& imc_config, march::ActuationMode mode,
-                                                               const urdf::JointConstSharedPtr& urdf_joint,
-                                                               march::PdoInterfacePtr pdo_interface,
-                                                               march::SdoInterfacePtr sdo_interface);
-  static std::unique_ptr<march::TemperatureGES> createTemperatureGES(const YAML::Node& temperature_ges_config,
-                                                                     march::PdoInterfacePtr pdo_interface,
-                                                                     march::SdoInterfacePtr sdo_interface);
-  static std::unique_ptr<march::PowerDistributionBoard>
+  std::shared_ptr<march::IMotionCube> createIMotionCube(const YAML::Node& imc_config, march::ActuationMode mode,
+                                                        const urdf::JointConstSharedPtr& urdf_joint,
+                                                        march::PdoInterfacePtr pdo_interface,
+                                                        march::SdoInterfacePtr sdo_interface);
+  std::shared_ptr<march::TemperatureGES> createTemperatureGES(const YAML::Node& temperature_ges_config,
+                                                              march::PdoInterfacePtr pdo_interface,
+                                                              march::SdoInterfacePtr sdo_interface);
+  std::shared_ptr<march::PowerDistributionBoard>
```

Comment:
Why has this been changed to a `shared_ptr`?

Reply:
Same